src/openassetio-core/src/hostApi/ManagerFactory.cpp

```diff
@@ -16,6 +17,11 @@
 #include <openassetio/typedefs.hpp>
 #include "openassetio/InfoDictionary.hpp"
 
+namespace {
+constexpr std::string_view kConfigDirVarNonBraced = "$config_dir";
```

Comment:
[Consider] I wonder if having an unbraced version is wise with this simplistic approach. It's unlikely other folks will be running substitutions themselves over this file, as some sort of preprocessing step, and even more unlikely that they would be using a conflicting token, (like $config_director_file_path or something), but it's possible.

This is probably fine, just thought i'd mention.

Reply:
I agree the prefix issue is real, and would require a more complete `find` to ensure the token is surrounded by a word boundary - probably at that point `std::regex` would be the way forward.

We did discuss the drawbacks of such a simple solution, with a view to fixing any issues only if/when someone cares enough.

To be fair, we didn't really consider the prefix issue, but it still seems like a very unlikely case.

I'm happy to look into to a more robust `std::regex` solution, though, if we think it's worthwhile?

Reply:
> Don't follow what you mean with the bracing not being unique to us,

I may have misinterpreted that you were also concerned about the general use of other var expansion methods, as well as the prefix-collision issue

Reply:
>concerned about the general use of other var expansion methods 

I was, but just in that it's a reason I could think of why folks might be putting `$token` formatted strings in this file.

> is it worth leaving the implementation as-is though

Emphatically yes, I regret mentioning this at all as I think it's just not worth the time of day we've given it.

Reply:
Just to be on the safe side in terms of 'as is' I was thinking of keeping the implementation structure, just dropping the brace-less input (vs rewriting to be simpler). But thats probably code smell 🙃

Reply:
Yeah I think we should get rid of the loop
```
for (const std::string_view& what : {kConfigDirVarNonBraced, kConfigDirVarBraced}) {
```
if there's only one element.

Reply:
Remove support for `$config_var` in 6d1ac38b